crates/ty_python_semantic/resources/mdtest/attributes.md

```diff
@@ -215,26 +215,41 @@ def get_int() -> int:
 def get_str() -> str:
     return "a"
 
+flag: bool = True
+
 class C:
+    global flag
+    if flag:
+        w: int
+    else:
+        w: str
+
     z: int
+    v: int = 1
 
     def __init__(self) -> None:
         self.x = get_int()
         self.y: int = 1
 
     def other_method(self):
         self.x = get_str()
-
-        # TODO: this redeclaration should be an error
         self.y: str = "a"
+        self.z: str = "a"
+        self.v: str = "a"
 
-        # TODO: this redeclaration should be an error
+    def anoter_method(self):
         self.z: str = "a"
 
 c_instance = C()
 
+# error: [conflicting-declarations] "Conflicting declared types for attribute `v`: `int` and `str`"
```

Comment:
I think it would be better if these diagnostics were emitted on (one of) the conflicting declarations, rather than on each usage site, and would be emitted regardless of whether the attribute is later accessed.

I realize this will likely require significant changes to the way this is implemented, because it means that in `TypeInferenceBuilder::check_class_definition` we would need to query for all declarations on the class and in methods, and error on the conflicting ones, rather than doing it lazily when looking up an attribute.

Reply:
@lipefree You marked this as resolved, but I don't think your current implementation emits the diagnostic on one of the conflicting declarations? Instead, it looks like the diagnostics are now emitted on the whole class declaration? A class declaration can be arbitrarily large, so it would be nice to attach it to one of the problematic declarations only.